app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1180,6 +1186,22 @@ class PluginApi {
   addFeaturedLinkMetaDecorator(decorator) {
     addFeaturedLinkMetaDecorator(decorator);
   }
+
+  addInOptionsForUsers(options) {
+    addInOptionsForUsers(options);
+  }
+
+  addInOptionsForAll(options) {
+    addInOptionsForAll(options);
+  }
+
+  addStatusOptions(options) {
+    addStatusOptions(options);
+  }
+
+  addPostTimeOptions(options) {
+    addPostTimeOptions(options);
+  }
```

Comment:
I think we need some more descriptive names here, because it's not clear they are related to advanced search.

`addAdvancedSearchInOptionsForUsers` would be better, but it is quite long. Do you think there is some way we could combine them into a single method like

```
addAdvancedSearchOptions({
  inOptionsForUsers: [
    ...my new options
  ],
  statusOptions: [
    ... other new options
  ]
})
```

We should also add some comments above the method with some usage examples.